kernel/src/arch/syscalls.c

```
@@ -327,17 +329,21 @@ int32_t syscall_spawn(struct regs *regs)
+    path[4] = '/';
     int i;
     for (i = 0; argv[0][i] && i < 122; i++)
-        path[i + 1] = argv[0][i];
-    path[i + 1] = '.';
-    path[i + 2] = 'b';
-    path[i + 3] = 'i';
-    path[i + 4] = 'n';
-    path[i + 5] = '\0';
+        path[i + 5] = argv[0][i];
+    path[i + 5] = '.';
+    path[i + 6] = 'b';
```

Cppcheck (2.21.0)

[error] Array 'path[128]' accessed at index 128, which is out of bounds.

(arrayIndexOutOfBounds)